_posts/2020-09-15-integrating-databricks-and-datadog.md

```diff
@@ -0,0 +1,176 @@
+---
+layout: post
+title: "Integrating Databricks jobs with Datadog"
+author: qphou
+tags:
+- featured
+- databricks
+- datadog
+- datapipe
+team: Core Platform
+---
+
+Batch and streaming Spark jobs are an integral part of our data platform and
+like our other production applications, we need
+[Datadog](https://datadoghq.com) instrumentation. We rely on
+[Databricks](https://databricks.com/customers/scribd) to power those Spark
+workloads, but integrating Datadog and Databricks wasn't turn-key. In this
+post, I'll share the two code snippets necessary to enable this integration: a custom cluster init script, and a special class to load into the Spark job.
+
+Rather than relying on the Spark UI in Databricks, piping these metrics into
+Datadog allows us to build extremely useful dashboards and more importantly
+**monitors** for our Spark workloads that can tie into our alerting
+infrastructure.
+
+
+## Configuring the Databricks cluster
+
+When creating the cluster in Databricks, we use the following init script-based
+configuration to set up the Datadog agent. It also likely possible to set this
+up via [customized containers with Databricks Container
+Services](https://docs.databricks.com/clusters/custom-containers.html) but the
+`databricks` runtime images don't get updated as frequently enough for our
+purposes.
+
+* Add cluster init script to setup datadog below
+* Set following environment variables for the cluster:
+  * `ENVIRONMENT=development/staging/production`
+  * `APP_NAME=your_spark_app_name`
+  * `DATADOG_API_KEY=KEY`
+
+All your Datadog metrics will be automatically tagged with `env` and `spark_app` tags.
+
+
+```bash
+#!/bin/bash
+# reference: https://docs.databricks.com/clusters/clusters-manage.html#monitor-performance
+#
+# This init script takes the following environment variables as input
+#   * DATADOG_API_KEY
+#   * ENVIRONMENT
+#   * APP_NAME
+
+echo "Setting up metrics for spark applicatin: ${APP_NAME}"
+echo "Running on the driver? $DB_IS_DRIVER"
+echo "Driver ip: $DB_DRIVER_IP"
+
+if [[ $DB_IS_DRIVER = "TRUE" ]]; then
+  cat << EOF >> /home/ubuntu/databricks/spark/conf/metrics.properties
+*.sink.statsd.host=${DB_DRIVER_IP}
+EOF
+
+  DD_INSTALL_ONLY=true \
+      DD_AGENT_MAJOR_VERSION=7 \
+      DD_API_KEY=${DATADOG_API_KEY} \
+      DD_HOST_TAGS="[\"env:${ENVIRONMENT}\", \"spark_app:${APP_NAME}\"]" \
+      bash -c "$(curl -L https://raw.githubusercontent.com/DataDog/datadog-agent/7.20.0-rc.10/cmd/agent/install_script.sh)"
```

Comment:
Nit: here is the latest stable version
```suggestion
      bash -c "$(curl -L https://raw.githubusercontent.com/DataDog/datadog-agent/7.22.0/cmd/agent/install_script.sh)"
```
There isn't a diff for the file in the comparison: https://github.com/DataDog/datadog-agent/compare/7.22.0...7.20.0-rc.10